### DISCUSSION TITLE
Background pdf

### DISCUSSION THREAD

**Discussion by ppiglets** [QuestPDF/QuestPDF#640] [Q&A] [answered]
Hello,

Is it possible to add an pdf as background (instead of a background image) to a page?

Paul

**Comment by girlpunk** [comment]
If the background is being created by QuestPDF, you can use layers to create the same effect

**Comment by DawidRobertNowak** [comment, marked as answer]
Hello,

We are closing this thread due to inactivity. We apologize for the delayed response. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Since this thread was opened, the [Document Operations](https://www.questpdf.com/concepts/document-operations.html) API (available from version 2024.12.0) added support for this. You can use an existing PDF as a page background by underlaying it beneath your generated content with UnderlayFile, and the RepeatSourcePages option lets a single background page repeat across every page of the document.

Should you encounter any further problems, please don't hesitate to open a new issue.
Thank you!
